deepchem/metalearning/torch_maml.py

```diff
@@ -0,0 +1,385 @@
+"""Model-Agnostic Meta-Learning (MAML) algorithm for low data learning."""
+
+import os
+import shutil
+import tempfile
+import time
+
+from deepchem.models.optimizers import Optimizer, Adam, GradientDescent, LearningRateSchedule
+from typing import Any, Dict, List, Optional, Tuple, Sequence, Union
+from deepchem.utils.typing import OneOrMany
+
+try:
+    from deepchem.metalearning import MetaLearner
+    import torch
+    has_pytorch = True
+except:
+    has_pytorch = False
+
+
+class MAML(object):
```

Comment:
This is also duplicated with Tensorflow. We should have a shared abstract class instead

Reply:
Documenting offline discussion, there is currently no abstract MAML superclass so we made a parallel torch implementation to existing TF. We may want to factor out an abstract class later